server/__tests__/unit/wallet-auth.merge-accounts.test.ts

```diff
@@ -0,0 +1,307 @@
+        'select nonce from account where id = $1',
+        [accountId],
+      );
+      const [{ nonce }] = query.rows;
+
+      // generate signature
+      const signature = genSignature(
+        userPrivKey,
+        userPubKey,
+        userAddr,
+        nonce,
+        genArbitraryConnectWalletData(nonce),
+      );
+
+      await expect(
+        mergeAccounts({
+          client,
+          signature,
+          accountId,
+          keepCurrentAccount: true,
+        }),
+      ).rejects.toThrow(
+        new UnauthorizedError('No account with the given wallet address'),
+      );
+    });
+  });
+  it('should throw an error if the account does not exist', async () => {
+    await withRootDb(async (client: PoolClient) => {
+      // inserting some account and delete it
+      const accountId = await createWeb2Account({ client, email });
+      const query = await client.query(
+        'select nonce from account where id = $1',
+        [accountId],
+      );
+      const [{ nonce }] = query.rows;
+      await client.query('DELETE FROM private.account where id = $1', [
+        accountId,
+      ]);
+      await client.query('DELETE FROM account where id = $1', [accountId]);
+
+      // inserting some web3 account
+      const { userPrivKey, userPubKey, userAddr } = await createNewUser();
+      await createAccountWithAuthUser(client, userAddr);
+
+      // generate signature
+      const signature = genSignature(
+        userPrivKey,
+        userPubKey,
+        userAddr,
+        nonce,
+        genArbitraryConnectWalletData(nonce),
+      );
+
+      await expect(
+        mergeAccounts({
+          client,
+          signature,
+          accountId,
+          keepCurrentAccount: true,
+        }),
+      ).rejects.toThrow(
+        new UnauthorizedError('Account not found for the given id'),
+      );
+    });
+  });
+  it('should throw an error if the web3 account also uses web2 login (email and/or google)', async () => {
+    await withRootDb(async (client: PoolClient) => {
+      const { accountId, signature, walletAccountId } =
+        await setupAccountsWithDataAccounts({
+          client,
+          email,
+          nameWeb2,
+          nameWeb3,
+        });
+
+      // set some email and google for the web3 account
+      await client.query(
+        'update private.account set email = $1, google_email = $2, google = $3 where id = $4',
+        [
+          'jane.doe@gmail.com',
+          'jane.doe@gmail.com',
+          'google123',
+          walletAccountId,
+        ],
+      );
+
+      await expect(
+        mergeAccounts({
+          client,
+          signature,
+          accountId,
+          keepCurrentAccount: true,
+        }),
+      ).rejects.toThrow(
+        new UnauthorizedError(
+          'You cannot connect your account to this wallet address. This wallet address is already associated with another email address.',
+        ),
+      );
+    });
+  });
+});
+
+async function checkAccountData({
```

Comment:
```suggestion
async function ensureAccountDataMigrated({
```

Hate to make this PR more complex but since we made the migration generic for any referenced data, the ideal tests would generate fake data in any referenceable table and then check that it's migrated. The current tests are very specific to the current set of tables that we have whereas the migration is generic. If this is too hard then it's fine, but I am a bit concerned by how risky and "destructive" this operation is.